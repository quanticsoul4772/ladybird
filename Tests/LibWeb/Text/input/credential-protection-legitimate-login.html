<!DOCTYPE html>
<html>
<head>
    <title>Credential Protection: Legitimate Login Flow Test</title>
    <script src="include.js"></script>
</head>
<body>
    <h1>Legitimate Login Test</h1>

    <!-- Legitimate form: same-origin submission -->
    <form id="legitForm" action="/login" method="POST">
        <input type="text" name="username" id="username" value="testuser" />
        <input type="password" name="password" id="password" value="testpass" />
        <button type="submit" id="submitBtn">Login</button>
    </form>

    <script>
        test(() => {
            println("=== Credential Protection: Legitimate Login Flow Test ===");
            println();

            // Test 1: Form exists and has correct attributes
            let form = document.getElementById('legitForm');
            println("Test 1: Form element exists");
            println("  form exists: " + (form !== null));
            println("  form.action: " + form.action);
            println("  form.method: " + form.method.toUpperCase());
            println();

            // Test 2: Form has password field
            let passwordField = document.getElementById('password');
            println("Test 2: Password field exists");
            println("  password field exists: " + (passwordField !== null));
            println("  password field type: " + passwordField.type);
            println("  password field has value: " + (passwordField.value.length > 0));
            println();

            // Test 3: Form action is same-origin
            let formActionURL = new URL(form.action);
            let currentOrigin = window.location.origin;
            println("Test 3: Form submission is same-origin");
            println("  current origin: " + currentOrigin);
            println("  form action origin: " + formActionURL.origin);
            println("  is same origin: " + (formActionURL.origin === currentOrigin));
            println();

            // Test 4: Verify form elements
            println("Test 4: Form has required elements");
            let usernameField = document.getElementById('username');
            println("  username field exists: " + (usernameField !== null));
            println("  username field name: " + usernameField.name);
            println("  password field name: " + passwordField.name);
            println("  form has 3 elements (including submit): " + (form.elements.length === 3));
            println();

            // Test 5: Protocol check
            println("Test 5: Protocol information");
            println("  current protocol: " + window.location.protocol);
            println("  form uses relative path: " + (form.action.startsWith('/login') || form.action.includes('/login')));
            println();

            println("=== Expected Behavior ===");
            println("For a legitimate same-origin form submission:");
            println("  - FormMonitor should detect the password field");
            println("  - FormMonitor should verify same-origin submission");
            println("  - NO security alert should be triggered");
            println("  - Form submission should proceed normally");
            println();

            println("=== Test Complete ===");
            println("This is a legitimate login scenario and should NOT trigger alerts.");
        });
    </script>
</body>
</html>
