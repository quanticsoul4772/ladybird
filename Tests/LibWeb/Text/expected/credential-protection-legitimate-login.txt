=== Credential Protection: Legitimate Login Flow Test ===
undefined
Test 1: Form element exists
  form exists: true
  form.action: file:///login
  form.method: POST
undefined
Test 2: Password field exists
  password field exists: true
  password field type: password
  password field has value: true
undefined
Test 3: Form submission is same-origin
  current origin: file://
  form action origin: file://
  is same origin: true
undefined
Test 4: Form has required elements
  username field exists: true
  username field name: username
  password field name: password
  form has 3 elements (including submit): true
undefined
Test 5: Protocol information
  current protocol: file:
  form uses relative path: true
undefined
=== Expected Behavior ===
For a legitimate same-origin form submission:
  - FormMonitor should detect the password field
  - FormMonitor should verify same-origin submission
  - NO security alert should be triggered
  - Form submission should proceed normally
undefined
=== Test Complete ===
This is a legitimate login scenario and should NOT trigger alerts.
